File: seleniumEasyJava/src/test/java/E2E/poms/progressBar/ProgressBarPOM.java
package E2E.poms.progressBar;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class ProgressBarPOM {
    private WebDriver driver;

    public ProgressBarPOM(WebDriver driver){
        this.driver = driver;

        PageFactory.initElements(driver, this);
    }

    // ------------------- Elements --------------------------------

    @FindBy(id = "cricle-btn")
    public WebElement downloadButton;

    @FindBy(className = "percenttext")
    public WebElement percentageText;
}

File: seleniumEasyPython/poms/progress_bar_POM.py
from selenium.webdriver.chrome.webdriver import WebDriver
from selenium.webdriver.common.by import By


class ProgressBarPOM:
    def __init__(self, driver: WebDriver):
        self.driver = driver

# ------------------- Elements --------------------------------

    def download_button(self):
        element = self.driver.find_element(By.ID, "cricle-btn")
        return element

    def percentage_text(self):
        element = self.driver.find_element(By.CLASS_NAME, "percenttext")
        return element
